### PR TITLE
Banner messages

### PR DESCRIPTION
**Pull request by tleraj**
The purpose of this PR is to fix https://github.com/aqualinkorg/aqualink-app/issues/324. The cases appear [here](https://docs.google.com/document/d/1JSioZbxul9gqILkUVTAyZAJknFVZ8Tv-0LiAgFfObkc/edit).

### CODE REVIEW

File: packages/website/src/routes/ReefRoutes/Reef/index.tsx [hunk 1]
```diff
@@ -40,30 +40,48 @@ const getAlertMessage = (
...
+
+    case status === "approved":
+      return "Application for an Aqualink Smart Buoy has been approved.";
+
+    case status === "rejected":
+      return (
+        <div>
+          Application for an Aqualink Smart Buoy not approved at this time. For
+          more information, you can contact<span> </span>
+          <a href="mailto: info@aqualink.org">info@aqualink.org</a>
```
Comment (ericboucher) [inline packages/website/src/routes/ReefRoutes/Reef/index.tsx:81]: no space after mailto:

File: packages/website/src/routes/RegisterSite/index.tsx [hunk 2]
```diff
@@ -137,6 +138,9 @@ const Apply = ({ classes }: ApplyProps) => {
             setNewReefId(data.reef.id);
             setDatabaseSubmissionOk(true);
             setSnackbarOpenFromDatabase(true);
+            if (user?.token) {
```
Comment (ericboucher) [inline packages/website/src/routes/RegisterSite/index.tsx:141]: why this change?

File: packages/website/src/routes/ReefRoutes/Reef/index.tsx [hunk 3]
```diff
@@ -40,30 +40,48 @@ const getAlertMessage = (
...
-          {defaultMessage} Apply for an Aqualink Smart Buoy
-          <span> </span> <Link to="/apply">here</Link>.
+          {defaultMessage} Your application for an Aqualink Smart Buoy is being
+          reviewed. You can check your application<span> </span>
+          <Link to="/apply">here</Link>.
         </div>
       );
+
+    case status === "approved":
+      return "Application for an Aqualink Smart Buoy has been approved.";
```
Comment (ericboucher) [inline packages/website/src/routes/ReefRoutes/Reef/index.tsx:74]: Let's say "Your application"

File: packages/website/src/routes/ReefRoutes/Reef/index.tsx [hunk 4]
```diff
@@ -40,30 +40,48 @@ const getAlertMessage = (
...
+
+    case status === "rejected":
+      return (
+        <div>
+          Application for an Aqualink Smart Buoy not approved at this time. For
```
Comment (ericboucher) [inline packages/website/src/routes/ReefRoutes/Reef/index.tsx:79]: Let's say "Your application..." for now

File: packages/website/src/routes/ReefRoutes/Reef/index.tsx (hunk 3 above)
Comment (tleraj) [reply packages/website/src/routes/ReefRoutes/Reef/index.tsx:74]: So the message should be `Your application for an Aqualink Smart Buoy has been approved.`?

File: packages/website/src/routes/RegisterSite/index.tsx (hunk 2 above)
Comment (tleraj) [reply packages/website/src/routes/RegisterSite/index.tsx:141]: We get the applications's info (`applied` and `status`) from the user's administered reefs. So we need to make sure that before we are redirected to the reef detail page, we have the up to date user's info. This wasn't the case, so I added this `dispatch(getSelf(user.token))` before we are redirected to the reef detail page either by the register page of by the application page.

**Review by ericboucher — APPROVED** [review]

**Review by ericboucher — APPROVED** [review]

File: packages/website/src/routes/ReefRoutes/Reef/Sensor/index.tsx [hunk 5]
```diff
@@ -26,20 +26,32 @@ import { userInfoSelector } from "../../../../store/User/userSlice";
...
-          Add a Smart Buoy
+          My Application
         </Link>
       );
+
+    case status === "approved":
+      return "Smart Buoy approved";
+
+    case status === "rejected":
+      return "Smart Buoy not approved";
```
Comment (ericboucher) [inline packages/website/src/routes/ReefRoutes/Reef/Sensor/index.tsx:54]: @tleraj can you maybe test the spacing for this in the app?

Comment (tleraj) [reply packages/website/src/routes/ReefRoutes/Reef/Sensor/index.tsx]: - Mobile

Screenshot from 2020-10-22 16-22-06


- Between 960px and 1220px:

Screenshot from 2020-10-22 16-21-20

- Wider screens
Screenshot from 2020-10-22 16-21-34
